src/index.js

```diff
@@ -3,51 +3,71 @@ import { createFilter } from '@rollup/pluginutils'
 export default function css(options = {}) {
   const filter = createFilter(options.include || ['**/*.css'], options.exclude)
   const styles = {}
-  const order = []
   let dest = options.output
-  let changes = 0
+  let hasChanged = false
+  let prevIds = []
 
   return {
     name: 'css',
+    buildStart() {
+      hasChanged = false
+    },
     transform(code, id) {
       if (!filter(id)) {
         return
       }
 
+      const { imports, codeWithoutImports } = splitImports(code);
+
       // When output is disabled, the stylesheet is exported as a string
       if (options.output === false) {
+        if (imports.length === 0) {
+          return {
+            code: `export default ['${JSON.stringify(code)}'`,
+            map: { mappings: '' }
+          }
+        }
+        const importNamed = imports.map((d, i) => `import i${i} from ${d}`).join('\n');
         return {
-          code: 'export default ' + JSON.stringify(code),
+          code: `
+            ${importNamed}
+            export default ${imports.map((_, i) => `i${i}`).join(' + ')} + '${JSON.stringify(codeWithoutImports)}'`,
           map: { mappings: '' }
         }
       }
 
-      // Track the order that each stylesheet is imported.
-      if (!order.includes(id)) {
-        order.push(id)
-      }
-
       // Keep track of every stylesheet
       // Check if it changed since last render
-      if (styles[id] !== code && (styles[id] || code)) {
-        styles[id] = code
-        changes++
+      // NOTE: If we are in transform block, we can assume styles[id] !== code, right?
```

Comment:
Can you verify this with the Rollup team? I thought transform was called when a file is saved, not modified per se.